admin/src/fqdn-to-provider/fqdn-to-provider.service.ts

```diff
@@ -35,119 +37,95 @@ export class FqdnToProviderService {
       },
     });
 
-    if (!fqdnToProvider) {
-      return [];
-    }
-
     return fqdnToProvider;
   }
 
-  async getProviderWithFqdns(
-    identityProvider: IIdentityProviderDTO,
-  ): Promise<IIdentityProviderDTO> {
-    const fqdnToProviders = await this.findFqdnsForOneProvider(
-      identityProvider.uid,
-    );
-
-    const idpWithFqdns = { ...identityProvider, fqdns: [] };
-    fqdnToProviders.forEach(({ fqdn }) => {
-      idpWithFqdns.fqdns.push(fqdn);
-    });
+  async getFqdnsForIdentityProviderUid(idpUid: string): Promise<string[]> {
+    const fqdnToProviders = await this.findFqdnsForOneProvider(idpUid);
 
-    return idpWithFqdns;
+    return fqdnToProviders.map(({ fqdn }) => fqdn);
   }
 
   async getProvidersWithFqdns(
-    identityProviders: IIdentityProvider[],
-  ): Promise<IIdentityProviderDTO[]> {
+    identityProviders: IdentityProviderFromDb[],
+  ): Promise<IdentityProviderWithFqdn[]> {
     const identityProvidersUids = identityProviders.map(
-      (idp: IIdentityProvider) => idp.uid,
+      (identityProviderFromDb) => identityProviderFromDb.uid,
     );
 
     const fqdnToProviders = await this.findFqdnsForProviders(
       identityProvidersUids,
     );
 
-    return this.getIdentityProvidersDTO(identityProviders, fqdnToProviders);
-  }
-
-  createFqdnsWithAcceptance(
-    fqdns: string[],
-  ): Array<Pick<IFqdnToProvider, 'fqdn' | 'acceptsDefaultIdp'>> {
-    return fqdns.map((fqdn) => {
-      return {
-        fqdn,
-        // by default, for now, the fqdn is accepted
-        acceptsDefaultIdp: true,
-      };
-    });
+    return this.aggregateFqdnToProviderWithIdentityProvider(
+      identityProviders,
+      fqdnToProviders,
+    );
   }
 
   async saveFqdnsProvider(
     identityProviderUid: string,
-    fqdns: Array<Pick<IFqdnToProvider, 'fqdn' | 'acceptsDefaultIdp'>>,
+    fqdns: string[],
   ): Promise<void> {
-    fqdns.filter(Boolean).forEach(async (fqdn) => {
+    const filteredFqdns = fqdns.filter(Boolean);
+    for (let index = 0; index < filteredFqdns.length; index++) {
```

Comment:
```suggestion
    for (const fqdn of filteredFqdns) {
```

Reply:
Applied in https://github.com/proconnect-gouv/federation/pull/424/files